sgx-tools/dcap_linux.go

```diff
@@ -0,0 +1,567 @@
+	rdata, err := proto.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	msgSize := uint32(len(rdata))
+	byteBuf := bytes.NewBuffer([]byte{})
+	binary.Write(byteBuf, binary.LittleEndian, &msgSize)
+	if _, err = conn.Write(byteBuf.Bytes()); err != nil {
+		return nil, err
+	}
+
+	if _, err = conn.Write(rdata); err != nil {
+		return nil, err
+	}
+
+	rdata = append(rdata[:rawMessageLength])
+	if _, err = conn.Read(rdata); err != nil {
+		return nil, err
+	}
+
+	byteBuf = bytes.NewBuffer(rdata)
+	if err = binary.Read(byteBuf, binary.LittleEndian, &msgSize); err != nil {
+		return nil, err
+	}
+
+	rdata = make([]byte, msgSize)
+	var msgSizeRead int
+	msgSizeRead, err = conn.Read(rdata)
+	if err != nil {
+		return nil, err
+	}
+
+	if msgSizeRead != int(msgSize) {
+		return nil, fmt.Errorf("invalid response size (returned %d, expected %d)",
+			msgSizeRead, msgSize)
+	}
+
+	return rdata, nil
+}
+
+func DumpTargetInfo(targetInfo []byte) error {
+	ti := &intelsgx.Targetinfo{}
+	if err := restruct.Unpack(targetInfo, binary.LittleEndian, &ti); err != nil {
+		return err
+	}
+
+	logrus.Debugf("Quoting Enclave's TARGETINFO:\n")
+	logrus.Debugf("  Enclave Hash:       0x%v\n",
+		hex.EncodeToString(ti.Measurement[:]))
+	logrus.Debugf("  Enclave Attributes: 0x%v\n",
+		hex.EncodeToString(ti.Attributes[:]))
+	logrus.Debugf("  CET Attributes:     %#02x\n",
+		ti.CetAttributes)
+	logrus.Debugf("  Config SVN:         %#04x\n",
+		ti.ConfigSvn)
+	logrus.Debugf("  Misc Select:        %#08x\n",
+		ti.MiscSelect)
+	logrus.Debugf("  Config ID:          0x%v\n",
+		hex.EncodeToString(ti.ConfigId[:]))
+
+	return nil
+}
+
+func DumpLocalReport(report []byte) error {
+	r := &intelsgx.Report{}
+	if err := restruct.Unpack(report, binary.LittleEndian, &r); err != nil {
+		return err
+	}
+
+	logrus.Debugf("REPORT:")
+	logrus.Debugf("  CPU SVN:                        0x%v\n",
+		hex.EncodeToString(r.CpuSvn[:]))
+	logrus.Debugf("  Misc Select:                    %#08x\n",
+		r.MiscSelect)
+	logrus.Debugf("  Product ID:                     0x%v\n",
+		hex.EncodeToString(r.IsvExtProdId[:]))
+	logrus.Debugf("  Attributes:                     0x%v\n",
+		hex.EncodeToString(r.Attributes[:]))
+	logrus.Debugf("  Enclave Hash:                   0x%v\n",
+		hex.EncodeToString(r.MrEnclave[:]))
+	logrus.Debugf("  Enclave Signer:                 0x%v\n",
+		hex.EncodeToString(r.MrSigner[:]))
+	logrus.Debugf("  Config ID:                      0x%v\n",
+		hex.EncodeToString(r.ConfigId[:]))
+	logrus.Debugf("  ISV assigned Produdct ID:       %#04x\n",
+		r.IsvProdId)
+	logrus.Debugf("  ISV assigned SVN:               %d\n",
+		r.IsvSvn)
+	logrus.Debugf("  Config SVN:                     %#04x\n",
+		r.ConfigSvn)
+	logrus.Debugf("  ISV assigned Product Family ID: 0x%v\n",
+		hex.EncodeToString(r.IsvFamilyId[:]))
+	logrus.Debugf("  Report Data:                    0x%v\n",
+		hex.EncodeToString(r.ReportData[:]))
+
+	return nil
+}
+
+func DumpEpidQuote(quote []byte) error {
```

Comment:
I suggest using DumpQuote to implement both epid and ecdsa quote dump, because the versions of them are different, so it is not necessary to implement 2 functions or 1 function with a so-called type parameter.

Reply:
> I suggest using DumpQuote to implement both epid and ecdsa quote dump, because the versions of them are different, so it is not necessary to implement 2 functions or 1 function with a so-called type parameter.

OK, thanks. The  functions about verification and dumping `quote` are not in the scope of this pr, these functions will be added later. So in this pr, I just delete the `DumpEpidQuote` function.